File: gcj2013QRB.hs
#!/usr/bin/env runghc
import Data.List
main = getContents >>= mapM_ putStrLn . zipWith caseno [1..] . solve . lines
caseno i r = "Case #"++show i++": "++r
solve (c:cs) = map (toString . answer) . take (read c) . analyze $ cs ::[String]
analyze (c:cs) = map (map (read::String->Int) . words) a: analyze r
  where (a,r) = splitAt (read . head . words $ c) cs
toString a = if a then "YES" else "NO"
answer a = and . zipWith f c $ a where
  f x = and . zipWith (\y->(>= min x y)) r
  c = map maximum a
  r = map maximum . transpose $ a

File: gcj2013QRB.py
#!/usr/bin/env python
import sys

def solve():
    N,M = map(int,sys.stdin.readline().split())
    a = [map(int,sys.stdin.readline().split()) for _ in range(N)]
    c = map(max,a)
    r = map(max,zip(*a))
    p = all(map(lambda x,ai: all(map(lambda y,aij: aij>=min(x,y),r,ai)),c,a))
    return "YES" if p else "NO"

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))
